src/main/java/jenkins/plugins/git/GitSCMBuilder.java

```diff
@@ -514,6 +514,9 @@ public GitSCM build() {
             extensions.add(new BuildChooserSetting(new AbstractGitSCMSource.SpecificRevisionBuildChooser(
                     (AbstractGitSCMSource.SCMRevisionImpl) revision)));
         }
+        if (head() instanceof GitRefSCMHead) {
+            withoutRefSpecs().withRefSpec(((GitRefSCMHead) head()).getRef());
```

Comment:
Now this seems a riskier place to clear out the existing ref specs. what is the justification for the trim of ref specs here